### PR TITLE
Update the guide to latest standards

### PR DESCRIPTION
**Pull request by jimmynotjim**
The content of the guide has not kept up with our everyday code
practices. This update brings the guide up to where we currently are as
a team.

## Changes

- Updated to include latest guidance on `use strict`
- Updated to enforce 2 space tab rules.
- Updated to include string and template rules.
- Updated to include the drop in interactive scripting to IE 9.

## Checklist

- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: standards/javascript.md [hunk 1]
```diff
@@ -109,21 +113,30 @@ to avoid potential issues with Automatic Semicolon Insertion _(ASI)_.
 
 ## Strings
 
-Strings should always be quoted using the same quotation mark. Use `'` or `"`
-consistently throughout your codebase. Ensure the team is using the same
-quotation mark in every portion of JavaScript that's authored. In general, we
-prefer to use single quotation marks `'`.
+- Use single quotes (`'`) for inline strings without variables or conditions.
+- Use [es6 emplate literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
```
Comment (contolini) [inline standards/javascript.md]: emplate -> template

File: standards/javascript.md [hunk 2]
```diff
@@ -133,16 +146,14 @@ far easier to format your strings, and the code looks a lot cleaner too.
 ## Variable Declaration
 
 Always declare variables in **a consistent manner**, and at the top of their
-scope. Keeping variable declarations to _one per line is encouraged_.
-Comma-first, a single `var` statement, multiple `var` statements, it's all
-fine, just be consistent across the project, and ensure the team is on the
-same page.
+scope, keeping variable declarations to single statments, including
```
Comment (contolini) [inline standards/javascript.md]: statments -> statements

File: standards/javascript.md [hunk 3]
```diff
@@ -109,21 +113,30 @@ to avoid potential issues with Automatic Semicolon Insertion _(ASI)_.
...
-prefer to use single quotation marks `'`.
+- Use single quotes (`'`) for inline strings without variables or conditions.
+- Use [es6 emplate literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
+  for inline strings with variables or conditions as well as multiline templates.
+- Use double quotes (`"`) for attribute values within templates.
 
 ##### Bad
 
 ```js
+const name = "Jane"
```
Comment (anselmbradford) [inline standards/javascript.md]: Add ending semicolon.

File: standards/javascript.md [hunk 4]
```diff
@@ -109,21 +113,30 @@ to avoid potential issues with Automatic Semicolon Insertion _(ASI)_.
...
+const input = "<input value='" + message + "'" +
+                     "type='text'" +
+                     "class='a-text-input'>";
 ```
 
 ##### Good
 
 ```js
-const message = 'oh hai ' + name + '!';
+const name = 'Jane'
```
Comment (anselmbradford) [inline standards/javascript.md]: Add ending semicolon.

File: standards/javascript.md [hunk 5]
```diff
@@ -287,6 +287,7 @@ instead of [function expressions][18]. Because [hoisting][19].
 const sum = ( x, y ) => {
   return x + y;
 };
+// Note for clarity, the issue is not arrow functions, but the expression
```
Comment (anselmbradford) [inline standards/javascript.md]: Needs an ending period?

**Comment by jimmynotjim** [comment]
@contolini @anselmbradford thanks for the reviews, fixed everything you found.

File: standards/javascript.md [hunk 6]
```diff
@@ -98,9 +99,12 @@ mark.
 - Symbolic constants should use `UPPERCASE`.
 - Encapsulated (private) variables and methods should use
   `_underscoreCamelCase`.
-- Prefix variables or properties that reference a jQuery object with `$` or
+- ~~Prefix variables or properties that reference a jQuery object with `$` or
   `_$`
-  (depending on exposure). For example: `var $button = $( '.btn-class' );`.
+  (depending on exposure). For example: `var $button = $( '.btn-class' );`.~~
+  _We are currently phasing out of our codebases. If you are working on new
```
Comment (Scotchester) [inline standards/javascript.md]: Missing the word `jQuery` here.

**Pull request merged**